src/scheduling/cost_models/simulated_quincy_cost_model.cc

```diff
@@ -0,0 +1,358 @@
+// The Firmament project
+// Copyright (c) 2014 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
+//
+// Quincy scheduling cost model, as described in the SOSP 2009 paper.
+
+#include "scheduling/cost_models/simulated_quincy_cost_model.h"
+
+#include <set>
+#include <string>
+#include <unordered_map>
+
+#include "base/common.h"
+#include "base/types.h"
+#include "misc/map-util.h"
+#include "misc/utils.h"
+#include "scheduling/common.h"
+#include "scheduling/cost_models/flow_scheduling_cost_model_interface.h"
+#include "scheduling/knowledge_base.h"
+
+namespace firmament {
+
+SimulatedQuincyCostModel::SimulatedQuincyCostModel(
+    shared_ptr<ResourceMap_t> resource_map,
+    shared_ptr<JobMap_t> job_map,
+    shared_ptr<TaskMap_t> task_map,
+    unordered_map<TaskID_t, ResourceID_t> *task_bindings,
+    unordered_set<ResourceID_t,
+      boost::hash<boost::uuids::uuid>>* leaf_res_ids,
+    KnowledgeBase* kb,
+    SimulatedDFS* dfs,
+    GoogleRuntimeDistribution *runtime_distribution,
+    GoogleBlockDistribution *block_distribution,
```

Comment:
I agree, but suggest we leave that change to a later PR and introduce a uniform abstraction across cost models. Most of the non-trivial ones take similar state-tracking arguments.